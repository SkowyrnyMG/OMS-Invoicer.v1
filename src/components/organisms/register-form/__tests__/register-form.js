import React from 'react';

import {
  renderWithReduxRouter,
  snapShotTest,
  screen,
  userEvent,
} from 'utils/tests/test-helper';
import RegisterForm from '../register-form';

describe('RegisterForm', () => {
  test('should display in the document with correct styles', () => {
    snapShotTest(<RegisterForm />);
  });

  test('should not submit the form if required fields are empty', async () => {
    renderWithReduxRouter(<RegisterForm />);

    const submitButton = screen.getByRole('button', { name: /submit/i });

    userEvent.click(submitButton);

    const errorMsg = await screen.findAllByText(/required.+/i);

    expect(errorMsg.length).not.toBeFalsy();
  });
});
