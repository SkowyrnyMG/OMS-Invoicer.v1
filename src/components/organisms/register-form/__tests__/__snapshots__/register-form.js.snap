// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterForm should display in the document with correct styles 1`] = `
.c3 {
  margin-bottom: 1rem;
  width: 100%;
  font-size: 14px;
  color: #F25C54;
}

.c1 {
  position: relative;
  display: block;
  width: 100%;
  margin-bottom: 3rem;
}

.c1:last-of-type {
  margin-bottom: 4rem;
}

.c1 input {
  border-color: #8F8F8F;
  -webkit-transition: 0.5s -webkit-transform;
  -webkit-transition: 0.5s transform;
  transition: 0.5s transform;
}

.c1 input:focus {
  -webkit-transform: translateX(2px);
  -ms-transform: translateX(2px);
  transform: translateX(2px);
}

.c1 input:placeholder-shown + div {
  opacity: 0;
  top: 50%;
}

.c1 input,
.c1 input::-webkit-input-placeholder {
  color: #8F8F8F;
  border-color: currentColor;
}

.c1 input,
.c1 input::-moz-placeholder {
  color: #8F8F8F;
  border-color: currentColor;
}

.c1 input,
.c1 input:-ms-input-placeholder {
  color: #8F8F8F;
  border-color: currentColor;
}

.c1 input,
.c1 input::placeholder {
  color: #8F8F8F;
  border-color: currentColor;
}

.c2 {
  position: absolute;
  top: -1rem;
  left: 0.3rem;
  font-size: 14px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  -webkit-transition: all 0.25s;
  transition: all 0.25s;
  pointer-events: none;
}

.c5 {
  padding: 0.8rem 1rem;
  width: 16rem;
  font-size: inherit;
  text-align: left;
  color: #FFFFFF;
  background-color: #446DF6;
  border: none;
  outline: none;
  border-radius: 0.5rem;
  cursor: pointer;
  -webkit-transition: 0.25s -webkit-transform;
  -webkit-transition: 0.25s transform;
  transition: 0.25s transform;
}

.c5:hover,
.c5:active {
  -webkit-transform: translateX(2px);
  -ms-transform: translateX(2px);
  transform: translateX(2px);
}

.c5:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

.c5::first-letter {
  text-transform: uppercase;
}

.c4 {
  margin: -2rem 0 2rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 45rem;
}

@media (max-width:767px) {
  .c5 {
    width: 100%;
  }
}

@media (max-width:680px) {
  .c0 {
    width: 100%;
    margin: 0 2rem;
  }
}

<form
  action="#"
  class="c0"
>
  <label
    class="c1"
    data-testid="formik-input"
    for="name"
  >
    <input
      id="name"
      name="name"
      placeholder="Your name"
      type="text"
      value=""
    />
    <div
      class="c2"
    >
      Your name
    </div>
    <span
      class="c3"
    />
  </label>
  <label
    class="c1"
    data-testid="formik-input"
    for="lastname"
  >
    <input
      id="lastname"
      name="lastname"
      placeholder="Your lastname"
      type="text"
      value=""
    />
    <div
      class="c2"
    >
      Your lastname
    </div>
    <span
      class="c3"
    />
  </label>
  <label
    class="c1"
    data-testid="formik-input"
    for="email"
  >
    <input
      id="email"
      name="email"
      placeholder="Your email"
      type="text"
      value=""
    />
    <div
      class="c2"
    >
      Your email
    </div>
    <span
      class="c3"
    />
  </label>
  <label
    class="c1"
    data-testid="formik-input"
    for="password"
  >
    <input
      id="password"
      name="password"
      placeholder="Your password"
      type="password"
      value=""
    />
    <div
      class="c2"
    >
      Your password
    </div>
    <span
      class="c3"
    />
  </label>
  <label
    class="c1"
    data-testid="formik-input"
    for="passwordConfirmation"
  >
    <input
      id="passwordConfirmation"
      name="passwordConfirmation"
      placeholder="Repeat password"
      type="password"
      value=""
    />
    <div
      class="c2"
    >
      Repeat password
    </div>
    <span
      class="c3"
    />
  </label>
  <div
    class="c4"
    data-testid="notification-popup"
  >
    <span
      class="c3"
      style="opacity: 0; visibility: hidden;"
    />
  </div>
  <button
    class="c5"
    type="submit"
  >
    Submit
  </button>
</form>
`;
